pages/details/[id].js

```diff
@@ -27,16 +29,30 @@ export default function Details() {
     });
   };
 
+  const onSubmitChangeDetail = (changedData) => {
+    setDestinations((destinations) => {
+      const updatedDestinations = destinations.map((destinationItem) => {
+        if (destinationItem.id === destination.id) {
+          return {
+            ...destinationItem,
+            startDate: changedData.changed === "Dates" ? changedData.startDate : destinationItem.startDate,
+            endDate: changedData.changed === "Dates" ? changedData.endDate : destinationItem.endDate,
```

Comment:
This can be simplified and made reusable - see previous comment. Also I would suggest a name like `update` or similar for what is called changedData now.

Reply:
Changed it to "updated".